### ISSUE TITLE
EF core 3.1 - Using Same table name in different databases

### ISSUE THREAD

**Issue by prathyushpalepu** [dotnet/efcore#24943]
I have an application which connects to multiple databases. I have separate DB context to connect to each database. I have my entities and fluent map classes in separate namespaces.  The issue I face is that both the databases have a table with the same name, and I run into the issue "Cannot use table 'dbo.User' for entity type 'Db2User' since it is being used for entity type 'Db1User' and there is no relationship between their primary keys". Since these are tables in separate databases, there is no relationship between their primary keys.  I have included my sample code below. 

Entity class for Database 1:
public class Db1User
    {
        public int Id { get; set; }

        public string Name { get; set; }
    }
Fluent Map for Database 1:
public class Db1UserMap : IEntityTypeConfiguration<Db1User>
    {
        public void Configure(EntityTypeBuilder<Db1User> builder)
        {
            builder.ToTable("User");
            builder.HasKey(x => x.Id);
            builder.Property(x => x.Id).HasColumnType("int").UseIdentityColumn().ValueGeneratedOnAdd().IsRequired();
            builder.Property(x => x.Name).HasColumnType("varchar").HasMaxLength(100).IsRequired();
        }
    }

Entity class for Database 2:
public class Db2User
    {
        public int Id { get; set; }

        public string ManagerName { get; set; }
    }

Fluent Map for Database 2:
public void Configure(EntityTypeBuilder<Db2User> builder)
        {
            builder.ToTable("User");
            builder.HasKey(x => x.Id);
            builder.Property(x => x.Id).HasColumnType("int").UseIdentityColumn().ValueGeneratedOnAdd().IsRequired();
            builder.Property(x => x.ManagerName).HasColumnType("varchar").HasMaxLength(100).IsRequired();
        }

DBContext for database 1:
public class Db1AppDbContext : DbContext
    {
        public Db1AppDbContext(DbContextOptions<Db1AppDbContext> options) : base(options)
        {
        }

        public virtual DbSet<Db1User> UserSet { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder
                .HasDefaultSchema("dbo")
                .ApplyConfigurationsFromAssembly(Assembly.GetAssembly(typeof(Db1AppDbContext)) ?? Assembly.GetCallingAssembly());
        }
    }

DBContext for database 2:
public class Db2AppDbContext : DbContext
    {
        public Db2AppDbContext(DbContextOptions<Db2AppDbContext> options) : base(options)
        {
        }

        public virtual DbSet<Db2User> UserSet { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder
                .HasDefaultSchema("dbo")
                .ApplyConfigurationsFromAssembly(Assembly.GetAssembly(typeof(Db2AppDbContext)) ?? Assembly.GetCallingAssembly());
        }
    }

EF Core version: 3.1.5
Database provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Target framework: (e.g. .NET Core 3.1)
Operating system: Windows 10
IDE: (e.g. Visual Studio 2019 16.3)

**Comment by AndriySvyryd** [comment]
@prathyushpalepu Put the `IEntityTypeConfiguration` classes into different assemblies or don't use `ApplyConfigurationsFromAssembly` to avoid mapping all entity types in both context types

**Comment by stevendarby** [comment]
There’s also an optional filter parameter when using `ApplyConfigurationsFromAssembly` which you might be able to use if there’s an easy way to identify which database each configuration type is for (e.g. type name starts with Db1).

**Comment by prathyushpalepu** [comment]
@AndriySvyryd @stevendarby Thank you for your suggestions. I was able to fix my issue using the predicate parameter on the ApplyConfigurationsFromAssembly method in 2 different ways. 
1. Since all the fluent maps of 1 database has the same namespace, I used the namespace as a filter when creating the model
2. I have created an empty base class per database, all fluent maps derive from the base class for that DB and filtered by the type of the class when creating the model
